tensorflow/python/keras/engine/compile_utils.py

```diff
@@ -209,7 +210,11 @@ def __call__(self,
         loss_metric_value *= ds_context.get_strategy().num_replicas_in_sync
 
       if batch_dim is None:
-        batch_dim = array_ops.shape(y_t)[0]
+        if tf_utils.is_ragged(y_t):
+          batch_dim = y_t.bounding_shape()[0]
```

Comment:
Please change to `batch_dim = y_t.nrows()` (to avoid unnecessary computation).

Reply:
Done.